components/Search/FuseSearchRecipe.js

```diff
@@ -0,0 +1,99 @@
+import AllRecipesList from "../AllRecipesList";
+import { useState, useEffect } from "react";
+import useSWR from "swr";
+import SearchBar from "./SearchBar";
+import styled from "styled-components";
+import Fuse from "fuse.js";
+
+const StyledUlBox = styled.div`
+  border: solid black 1px;
+  padding: 10px;
+  width: 500px;
+  margin: 20px;
+`;
+
+const fuseOptions = {
+  threshold: 0.3,
+  keys: ["title"],
+};
+
+export default function FuseSearchRecipe() {
+  const [results, setResults] = useState([]);
+  const [fuse, setFuse] = useState(null);
+  const [isFuseActive, setIsFuseActive] = useState(false);
+  const [recipes, setRecipes] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    async function getRecipes() {
+      try {
+        const response = await fetch(`/api/recipes`);
+        const fetchedRecipes = await response.json();
+        setRecipes(fetchedRecipes);
+        setIsLoading(false);
+        setFuse(new Fuse(fetchedRecipes, fuseOptions));
+      } catch (error) {
+        setError(error);
+        setIsLoading(false);
+      }
+    }
+    getRecipes();
+  }, []);
```

Comment:
If an error occurs, you currently don't reset it anywhere.

I would recommend changing your code to this:

```suggestion
useEffect(() => {
  async function getRecipes() {
    // Reset the error state at the beginning of each call
    setError(null);
    setIsLoading(true);

    try {
      const response = await fetch(`/api/recipes`);
      const fetchedRecipes = await response.json();

      setRecipes(fetchedRecipes);
      setFuse(new Fuse(fetchedRecipes, fuseOptions));
    } catch (fetchError) {
      setError(fetchError);
    } finally {
      // Always turn off loading indicator, regardless of the outcome
      setIsLoading(false);
    }
  }

  getRecipes();
}, []);
```